main.go

```diff
@@ -382,6 +375,35 @@ func singleflightDo[V any](group *singleflight.Group, key string, fn func() (V,
 	return out.(V), err, shared
 }
 
+// passthroughHandler is an HTTP handler that passes through GET requests to the CT log.
+type passthroughHandler struct {
+	logURL string
+}
+
+func (p passthroughHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		fmt.Fprintln(w, "only GET is supported")
+		return
+	}
+	url := fmt.Sprintf("%s%s", p.logURL, r.URL.Path)
+	r, err := http.NewRequestWithContext(r.Context(), http.MethodGet, url, nil)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, "creating request: %s\n", err)
+		return
+	}
+	resp, err := http.DefaultClient.Do(r)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, "fetching %s: %s\n", url, err)
+		return
+	}
```

Comment:
I think `resp.Body` needs to be closed before we return.

```suggestion
	}
	defer resp.Body.Close()
```